webapp/src/js/app.js

```diff
@@ -107,6 +107,31 @@ _.templateSettings = {
     return;
   }
 
+  var MAIN_ROUTES = ['tasks', 'messages', 'contacts', 'analytics', 'reports'];
+
+  var matchMainRoute = function(name) {
+    for(var i=0; i < MAIN_ROUTES.length; i++) {
+      if(name.indexOf(MAIN_ROUTES[i])>=0) {
+        return MAIN_ROUTES[i];
+      }
+    }
+    return null;
+  };
+
+  // Detects reloads or route updates (#/something)
+  app.run(function($rootScope, $state, Session, Auth) {
+    $rootScope.$on('$stateChangeStart', function(event, toState) {
+      if(Session && toState.name.indexOf('error')===-1) {
+        var route = matchMainRoute(toState.name);
+        if(route) {
+          Auth.any([['can_access_'+route]]).catch(function() {
```

Comment:
It's a little cleaner to just treat the Auth service as a function, eg: `Auth('can_access_tasks').then(...`.

I don't like concatenating things like permissions because if we want to say find and replace "can_access_tasks" then we'll miss this one. Instead can `MAIN_ROUTES` be a map of route prefix to required permission? eg:

```js
  var ROUTE_PERMISSIONS = {
    tasks: 'can_access_tasks',
    ...
  };
  var getRequiredPermission = function(route) {
    var baseRoute = route.split('.')[0];
    return ROUTE_PERMISSIONS[baseRoute];
  };
```